### PR TITLE
Local device info object for pair/unpair

### PR DESCRIPTION
**Pull request by harshdeepsehgal**
Added a local `DeviceInformation` object to make sure flags are updated to enable pairing and unpairing.

Solves #699

### CODE REVIEW

File: bleak/backends/winrt/client.py [hunk 1]
```diff
@@ -406,10 +408,10 @@ async def unpair(self) -> bool:
             Boolean on whether the unparing was successful.
 
         """
-
-        if self._requester.device_information.pairing.is_paired:
+        device_information=await DeviceInformation.create_from_id_async(self._requester.device_information.id)
```
Comment (dlech) [inline bleak/backends/winrt/client.py:411]: Can we add a comment here explaining that the old `device_information` is out of date?

File: bleak/backends/winrt/client.py [hunk 2]
```diff
@@ -395,7 +397,7 @@ def handler(sender, args):
                 )
                 return True
         else:
-            return self._requester.device_information.pairing.is_paired
+            return device_information.pairing.is_paired
```
Comment (dlech) [inline bleak/backends/winrt/client.py:400]: Should we just return `False` here?

**Review by dlech — COMMENTED** [review]
Thanks!

Can you run `black` on the code to fix the formatting and add a changelog entry?

File: bleak/backends/winrt/client.py (hunk 2 above)
Comment (dlech) [reply bleak/backends/winrt/client.py:400]: Never mind, I see how this could be `True` now.

File: bleak/backends/winrt/client.py (hunk 1 above)
Comment (harshdeepsehgal) [reply bleak/backends/winrt/client.py]: Done

**Comment by harshdeepsehgal** [comment]
I have added the changelog entry. I did run black, it has reformatted files, but somehow it still fails the check.

**Comment by dlech** [comment]
Black just came out with a new stable version. The CI is using `black-22.1.0`. Maybe you are using an older version? Or maybe we haven't run the new version on the code yet?

**Comment by harshdeepsehgal** [comment]
Yes I was running an older version of black. It passed the checks now.

File: bleak/backends/corebluetooth/characteristic.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def __init__(self, obj: CBCharacteristic):
         # self.__props = obj.properties()
         self.__props: List[str] = [
             _GattCharacteristicsPropertiesEnum[v][0]
-            for v in [2 ** n for n in range(10)]
+            for v in [2**n for n in range(10)]
```
Comment (dlech) [inline bleak/backends/corebluetooth/characteristic.py:66]: It would be nice to put unrelated black format changes in a separate commit.

File: bleak/backends/winrt/client.py [hunk 4]
```diff
@@ -350,15 +351,18 @@ async def pair(self, protection_level: int = None, **kwargs) -> bool:
             Boolean regarding success of pairing.
 
         """
-
+        # New local device information oject created since the object from the requester isn't updated
```
Comment (dlech) [inline bleak/backends/winrt/client.py:354]: spelling

```suggestion
        # New local device information object created since the object from the requester isn't updated
```

File: bleak/backends/winrt/client.py [hunk 5]
```diff
@@ -407,10 +411,12 @@ async def unpair(self) -> bool:
 
         """
 
-        if self._requester.device_information.pairing.is_paired:
-            unpairing_result = (
-                await self._requester.device_information.pairing.unpair_async()
-            )
+        # New local device information oject created since the object from the requester isn't updated
```
Comment (dlech) [inline bleak/backends/winrt/client.py:414]:
```suggestion
        # New local device information object created since the object from the requester isn't updated
```

File: bleak/backends/corebluetooth/characteristic.py (hunk 3 above)
Comment (dlech) [reply bleak/backends/corebluetooth/characteristic.py:66]: Actually, I'll just go ahead and do that, then you can just rebase on the develop branch.

**Comment by dlech** [comment]
Thanks!

**Comment by harshdeepsehgal** [comment]
Happy to contribute :)
